eth/catalyst/simulated_beacon.go

```diff
@@ -144,40 +143,49 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
 	feeRecipient := c.feeRecipient
 	c.feeRecipientLock.Unlock()
 
+	// Reset to CurrentBlock in case of the chain was rewound
+	if header := c.eth.BlockChain().CurrentBlock(); c.curForkchoiceState.HeadBlockHash != header.Hash() {
+		finalizedHash := c.finalizedBlockHash(header.Number.Uint64())
+		c.setCurrentState(header.Hash(), *finalizedHash)
```

Comment:
`finalizedHash` can be nil, should have nil check here.

Reply:
Were you able to trigger a panic here?

It could in theory but should be extremely unlikely:  a setHead which rolls back to before the latest finalized block would have to occur between the execution of line 147 and 148.